addons/calendar/static/src/js/services/calendar_notification_service.js

```diff
@@ -0,0 +1,121 @@
+/** @odoo-module **/
+
+import { serviceRegistry } from "@web/webclient/service_registry";
+import { browser } from "@web/core/browser";
+import { ConnectionLostError } from "@web/services/rpc_service";
+
+export const calendarNotificationService = {
+
+    async deploy(env) {
+
+        let _calendarNotifTimeouts = {};
+        let _getNextCalendarNotifTimeout = null;
+        let _calendarNotif = {};
+
+        env.bus.on('WEB_CLIENT_READY', null, async () => {
+            const legacyEnv = owl.Component.env;
+            legacyEnv.services.bus_service.onNotification(this, _onNotification);
+            legacyEnv.services.bus_service.startPolling();
+        });
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * Displays the Calendar notification on user's screen
+         *
+         * @private
+         */
+        function _displayCalendarNotification(notifications) {
+            var lastNotifTimer = 0;
+
+            // Clear previously set timeouts and destroy currently displayed calendar notifications
+            browser.clearTimeout(_getNextCalendarNotifTimeout);
+            _.each(_calendarNotifTimeouts, browser.clearTimeout);
+            _calendarNotifTimeouts = {};
+
+            // For each notification, set a timeout to display it
+            _.each(notifications, function (notif) {
```

Comment:
lodash forEach is ok ? or native forEach is better ?

Reply:
prefer native always :)

Reply:
we don't use _ anymore, so forEach, or for ()